debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/sink/SinkConnection.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.sink;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.kafka.common.config.Config;
+import org.apache.kafka.common.config.ConfigValue;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.MongoSecurityException;
+import com.mongodb.ReadPreference;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.event.ClusterClosedEvent;
+import com.mongodb.event.ClusterDescriptionChangedEvent;
+import com.mongodb.event.ClusterListener;
+import com.mongodb.event.ClusterOpeningEvent;
+
+import io.debezium.config.Field;
+
+public final class SinkConnection {
```

Comment:
Could this be refactored to not use just static methods?

Reply:
@jpechane Hmmm, I think a better class name would be `SinkMongoClientFactory` or something instead. This class/instance does not really reflect anymore its initial intention. wdyt?

Reply:
There is one huge caveat in creating new layer of connection management for sink -- even considering the Source one needs some refactoring. You are losing the ability to use SSL connection. 

Java client for Mongo doesn't support client certificate configuration options (so things like `tlsCertificateKeyFile` etc...) and these need to be handled programatically...

Reply:
Agreed, and I still need to file a Jira to add SSL support as it might be important for many (cloud) scenarios.
@jcechace so what is your final conclusion? Keep it (maybe renamed) with static intializers or start a refactoring to a non-static/real `SinkConnection` class similar to the ones in the source connectors?  I need to finish this initial implementation PR this week because my PTO comes up then.

Reply:
@rk3rn3r TBH it's up to @jpechane  in what form he is willing to merge. I personally think that SSL support is important and thus we should  consolidate the connection -- e.g. if possible reuse the one from source now and extend / refactor later. However since we are now in the 3.0 development cycle anyway I don't have a strong objection against doing this change in several PRs.

Reply:
It is not ideal but I am willing to postpone it for full refactoring.